sdks/python/apache_beam/runners/worker/operations.py

```diff
@@ -886,7 +886,7 @@ def output_key(self, wkey, accumulator):
     if windows is 0:
       self.output(_globally_windowed_value.with_value((key, value)))
     else:
-      self.output(WindowedValue((key, value), windows[0].end, windows))
+      self.output(WindowedValue((key, value), windows[0].max_timestamp(), windows))
```

Comment:
Would not this make the timestamp for the element incorrect?

Reply:
Why is it any more incorrect than "end"?  max_timestamp = end - 0.001

Reply:
You are right. I read this as just 'max_timestamp()', did not realize it is the max timestamp for the window. Your change looks correct.

Referencing your email (https://lists.apache.org/thread.html/ad42c55ed0212cb18b2b29bfc3dddfb47b8cb9f3358583775d0da37b@%3Cdev.beam.apache.org%3E), this needs to be a shared definition between Python and Java. @lukecwik could you confirm that an element in a window should have a timestamp < window.end or could it be <= window.end ?

Reply:
This change certainly fixes a bug (though I agree the docs could and should be cleaned up). 

We should file a JIRA that we're not supporting more general timestamp combining fns here.